src/realm/db.cpp

```diff
@@ -2249,6 +2249,7 @@ void DB::low_level_commit(uint_fast64_t new_version, Transaction& transaction)
                 // mode the file on disk may very likely be in an invalid state.
                 break;
         }
+        this->m_file_format_version = transaction.get_file_format_version();
```

Comment:
Why `this->`?

Reply:
Because then my IDE would help me with the member name :)

Reply:
This is problematic wrt the rules for automated backup.  Making a backup is tied to file format upgrade. Either we have to break that coupling, or we'll suddenly trigger a complete backup as part of a commit - which 1) may freeze the app for quite some time and 2) vastly increases the complexity of the backup machinery, as it now has to work as part of commit instead of session initiation.